vllm/model_executor/layers/quantization/quark/schemes/quark_w4a4_mxfp4.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Callable, Dict, List, Optional
+
+import torch
+import torch.nn.functional as F
+
+from vllm.model_executor.layers.quantization.quark.schemes import QuarkScheme
+from vllm.model_executor.parameter import (GroupQuantScaleParameter,
+                                           PackedvLLMParameter)
+from vllm.platforms import current_platform
+
+from vllm.model_executor.layers.quantization.utils.mxfp4_utils import OCP_MX_BLOCK_SIZE
+
+__all__ = ["QuarkW4A4MXFP4"]
+
+
+
+
+class QuarkW4A4MXFP4(QuarkScheme):
+
+    def __init__(self, weight_quant_spec: Dict[str, Any],
+                 input_quant_spec: Dict[str, Any]):
+        self.out_dtype = torch.get_default_dtype()
+        self.qscheme = "per_group"
+        self.weight_quant_spec = weight_quant_spec
+        self.input_quant_spec = input_quant_spec
+        self.emulate = not current_platform.supports_mx()
+
+        if self.emulate:
+            try:
+                from quark.torch.export.nn.modules import realquantizer
+                from quark.torch.quantization.config.config import (
+                    QuantizationSpec)
+            except ImportError as err:
+                raise ImportError(
+                    "The package `amd-quark` is required to use AMD Quark "
+                    "MX-FP4 models. Please install it with `pip install "
+                    "amd-quark`.") from err
+
+            input_quant_spec = QuantizationSpec.from_dict(
+                self.input_quant_spec)
+
+            self.input_quantizer = realquantizer.get_real_quantizer(
+                qspec=input_quant_spec,
+                quantizer=None,
+                real_quantized=False,
+                float_dtype=self.out_dtype,
+            )
+
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        # lovelace and up
+        return 89
```

Comment:
seems like this should be 100 for blackwell and up (not sure if AMD uses these numbers as well)

Reply:
updated and lowered the number. We support running in emulation for below blackwell (and AMD similar).